commons/src/main/java/org/infinispan/commons/marshall/MarshallUtil.java

```diff
@@ -63,13 +65,59 @@
       if (size == NULL_VALUE) {
          return null;
       }
-      final T map = Objects.requireNonNull(builder, "MapBuilder must be non-null").build(size);
+      final T map = Objects.requireNonNull(builder, "MapBuilder must be non-null").build(computeCapacity(size));
       for (int i = 0; i < size; i++) //noinspection unchecked
          map.put((K) in.readObject(), (V) in.readObject());
       return map;
    }
 
    /**
+    * Marshall the {@code map} to the {@code ObjectOutput}.
+    * <p>
+    * {@code null} maps are supported.
+    *
+    * @param map {@link Map} to marshall.
+    * @param out {@link ObjectOutput} to write. It must be non-null.
+    * @param <K> Key type of the map.
+    * @param <V> Value type of the map.
+    * @param <T> Type of the {@link Map}.
+    * @throws IOException If any of the usual Input/Output related exceptions occur.
+    */
+   public static <K, V, T extends Map<K, V>> void marshallMap(T map, ElementWriter<K> keyWriter, ElementWriter<V> valueWrite, ObjectOutput out) throws IOException {
```

Comment:
True, you'd have to change all the collection marshall methods, so let's leave it as is.